pages/orders/[orderId]/modal.tsx

```diff
@@ -0,0 +1,124 @@
+  grid-template: 'dt dd' auto / 1fr auto;
+  grid-auto-rows: auto;
+  line-height: ${({ theme }) => theme.lineHeight.medium};
+  margin: 0;
+
+  dt {
+    grid-area: 'dt';
+  }
+
+  dd {
+    grid-area: 'dd';
+    text-align: right;
+  }
+`;
+
+const StyledSpan = styled.span`
+  font-weight: ${({ theme }) => theme.typography.fontWeight.semiBold};
+`;
+
+const InternalOrderModalPage: React.FC<{ order: Order }> = ({ order }) => {
+  const { billing_address } = order;
+
+  const formatCurrency = (amount: string) =>
+    new Intl.NumberFormat(order.customer_locale, { style: 'currency', currency: order.currency_code }).format(parseFloat(amount));
+
+  return (
+    <Grid gridColumns="repeat(auto-fill, minmax(16rem, 1fr))" gridGap="3rem">
+      <GridItem>
+        <H3>Billing information</H3>
+        <StyledAddress>
+          <div>
+            {billing_address.first_name} {billing_address.last_name}
+          </div>
+          <div>{billing_address.street_1}</div>
+          {billing_address.street_2 && <div>{billing_address.street_2}</div>}
+          <div>
+            {billing_address.city}, {billing_address.state}, {billing_address.zip}
+          </div>
+          <div>{billing_address.country}</div>
+        </StyledAddress>
+      </GridItem>
+      <GridItem>
+        <Flex
+          alignItems="center"
+          flexDirection="row"
+          flexWrap="nowrap"
+          justifyContent="space-between"
+          marginBottom="small"
+        >
+          <H3 marginBottom="none">Payment details</H3>
+          <Badge label={order.payment_status} variant="success" />
+        </Flex>
+        <StyledDl>
+          <dt>Subtotal</dt>
+          <dd>{formatCurrency(order.subtotal_ex_tax)}</dd>
+          <dt>Discount</dt>
+          <dd>-{formatCurrency(order.discount_amount)}</dd>
+          <dt>Shipping</dt>
+          <dd>{formatCurrency(order.shipping_cost_ex_tax)}</dd>
+          <dt>Tax</dt>
+          <dd>{formatCurrency(order.total_tax)}</dd>
+          <dt>
+            <StyledSpan>Grand total</StyledSpan>
+          </dt>
+          <dd>
+            <StyledSpan>{formatCurrency(order.total_inc_tax)}</StyledSpan>
+          </dd>
+        </StyledDl>
+      </GridItem>
+      <GridItem>
+        <H3>Order information</H3>
+        <StyledDl>
+          <dt>ID</dt>
+          <dd>{order.id}</dd>
+          <dt>Type</dt>
+          <dd>
+            <span style={{ textTransform: 'capitalize' }}>{order.order_source}</span>
+          </dd>
+          <dt>Status</dt>
+          <dd>{order.status}</dd>
+          <dt>Total items</dt>
+          <dd>
+            {order.items_total > 1 ? `${order.items_total} items` : `${order.items_total} item`}
+          </dd>
+        </StyledDl>
+      </GridItem>
+    </Grid>
+  );
+};
+
+const OrderModalPage: React.FC = () => {
+  const router = useRouter();
+  const { orderId } = router.query;
+  const { isLoading, order, error } = useOrder(parseInt(`${orderId}`, 10));
+
+  if (isLoading) {
```

Comment:
🍹 totally optional, but mention for consistency - on our other pages we're using a bit of modern JS to place this on one line.

e.g.
```jsx
if (isLoading) return null;
```

Reply:
IMO, not a fan 😅 This plays into your comment below, but we should add eslint if we want consistency like this. Most of these files I've copied into a repo using our eslint config and pasted them back here.

Reply:
This isn't "modern" though, this pattern has existed for as long as JS. It was pretty popular in the beginning of JS (and other languages). It is not recommended though, it goes against our code style guide and it can lead to issues. It also makes the diff worse when needing to add new stuff to the if block.

```diff
-if (isLoading) return null;
+if (isLoading) {
+  console.log('something else');
+  return null;
+}
```

vs:

```diff
if (isLoading) {
+  console.log('something else');
  return null;
}
```